### PR TITLE
fix: ensure SimpleFeatureStateViewSet uses correct permissions for segment overrides

### PR DESCRIPTION
**Pull request by matthewelwell**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

Fixes an issue found when testing the new MANAGE_SEGMENTS permission functionality. The FE uses the SimpleFeatureStateViewSet for updating segment overrides, which the new permission didn't cover. This PR ensures that this endpoint is correctly covered by the new permission for creating (although this is not used by the FE) and updating segment overrides via this API route. 

## How did you test this code?

Added new unit tests.

### CODE REVIEW

File: api/tests/unit/features/test_unit_features_views.py [hunk 1]
```diff
@@ -1264,3 +1270,153 @@ def test_cannot_update_feature_of_a_feature_state(
     assert (
         response.json()["feature"][0] == "Cannot change the feature of a feature state"
     )
+
+
+def test_create_segment_override__using_simple_feature_state_viewset__allows_manage_segment_overrides(
+    staff_client: APIClient,
+    with_environment_permissions: Callable[[list[str], int], None],
```
Comment (khvn26) [inline api/tests/unit/features/test_unit_features_views.py]:
```suggestion
    with_environment_permissions: Callable[[list[str]], None],
```

Not precisely accurate but correct

File: api/tests/unit/features/test_unit_features_views.py [hunk 2]
```diff
@@ -1264,3 +1270,153 @@ def test_cannot_update_feature_of_a_feature_state(
...
+    assert response.status_code == status.HTTP_201_CREATED
+
+    assert FeatureState.objects.filter(
+        feature=feature, environment=environment, feature_segment=feature_segment
+    ).exists()
+
+
+def test_create_segment_override__using_simple_feature_state_viewset__denies_update_feature_state(
+    staff_client: APIClient,
+    with_environment_permissions: Callable[[list[str], int], None],
```
Comment (khvn26) [inline api/tests/unit/features/test_unit_features_views.py]:
```suggestion
    with_environment_permissions: Callable[[list[str]], None],
```

File: api/tests/unit/features/test_unit_features_views.py [hunk 3]
```diff
@@ -1264,3 +1270,153 @@ def test_cannot_update_feature_of_a_feature_state(
...
+        url, data=json.dumps(data), content_type="application/json"
+    )
+
+    # Then
+    assert response.status_code == status.HTTP_403_FORBIDDEN
+
+
+def test_update_segment_override__using_simple_feature_state_viewset__allows_manage_segment_overrides(
+    staff_client: APIClient,
+    with_environment_permissions: Callable[[list[str], int], None],
```
Comment (khvn26) [inline api/tests/unit/features/test_unit_features_views.py]:
```suggestion
    with_environment_permissions: Callable[[list[str]], None],
```

File: api/tests/unit/features/test_unit_features_views.py [hunk 4]
```diff
@@ -1264,3 +1270,153 @@ def test_cannot_update_feature_of_a_feature_state(
...
+        environment=environment,
+        feature_segment=feature_segment,
+        enabled=True,
+        feature_state_value__string_value="foo",
+    ).exists()
+
+
+def test_update_segment_override__using_simple_feature_state_viewset__denies_update_feature_state(
+    staff_client: APIClient,
+    with_environment_permissions: Callable[[list[str], int], None],
```
Comment (khvn26) [inline api/tests/unit/features/test_unit_features_views.py]:
```suggestion
    with_environment_permissions: Callable[[list[str]], None],
```

File: api/tests/unit/features/test_unit_features_views.py [hunk 5]
```diff
@@ -1264,3 +1270,153 @@ def test_cannot_update_feature_of_a_feature_state(
...
+
+
+def test_create_segment_override__using_simple_feature_state_viewset__allows_manage_segment_overrides(
+    staff_client: APIClient,
+    with_environment_permissions: Callable[[list[str], int], None],
+    environment: Environment,
+    feature: Feature,
+    segment: Segment,
+    feature_segment: FeatureSegment,
+):
```
Comment (khvn26) [inline api/tests/unit/features/test_unit_features_views.py]:
```suggestion
) -> None:
```

File: api/tests/unit/features/test_unit_features_views.py [hunk 6]
```diff
@@ -1264,3 +1270,153 @@ def test_cannot_update_feature_of_a_feature_state(
...
+
+
+def test_create_segment_override__using_simple_feature_state_viewset__denies_update_feature_state(
+    staff_client: APIClient,
+    with_environment_permissions: Callable[[list[str], int], None],
+    environment: Environment,
+    feature: Feature,
+    segment: Segment,
+    feature_segment: FeatureSegment,
+):
```
Comment (khvn26) [inline api/tests/unit/features/test_unit_features_views.py]:
```suggestion
) -> None:
```

File: api/tests/unit/features/test_unit_features_views.py [hunk 7]
```diff
@@ -1264,3 +1270,153 @@ def test_cannot_update_feature_of_a_feature_state(
...
+
+def test_update_segment_override__using_simple_feature_state_viewset__allows_manage_segment_overrides(
+    staff_client: APIClient,
+    with_environment_permissions: Callable[[list[str], int], None],
+    environment: Environment,
+    feature: Feature,
+    segment: Segment,
+    feature_segment: FeatureSegment,
+    segment_featurestate: FeatureState,
+):
```
Comment (khvn26) [inline api/tests/unit/features/test_unit_features_views.py]:
```suggestion
) -> None:
```

File: api/tests/unit/features/test_unit_features_views.py [hunk 8]
```diff
@@ -1264,3 +1270,153 @@ def test_cannot_update_feature_of_a_feature_state(
...
+
+def test_update_segment_override__using_simple_feature_state_viewset__denies_update_feature_state(
+    staff_client: APIClient,
+    with_environment_permissions: Callable[[list[str], int], None],
+    environment: Environment,
+    feature: Feature,
+    segment: Segment,
+    feature_segment: FeatureSegment,
+    segment_featurestate: FeatureState,
+):
```
Comment (khvn26) [inline api/tests/unit/features/test_unit_features_views.py]:
```suggestion
) -> None:
```

File: api/tests/unit/features/test_unit_features_views.py (hunk 1 above)
Comment (matthewelwell) [reply api/tests/unit/features/test_unit_features_views.py]: Well, technically, I guess it should be `Callable[[list[str], int | None], None]` based on the definition: 

```python
@pytest.fixture()
def with_environment_permissions(
    environment: Environment, staff_user: FFAdminUser
) -> typing.Callable[[list[str], int], None]:
    """
    Add environment permissions to the staff_user fixture.
    Defaults to associating to the environment fixture.
    """

    def _with_environment_permissions(
        permission_keys: list[str], environment_id: typing.Optional[int] = None
    ) -> UserEnvironmentPermission:
        environment_id = environment_id or environment.id
        uep, __ = UserEnvironmentPermission.objects.get_or_create(
            environment_id=environment_id, user=staff_user
        )
        uep.permissions.add(*permission_keys)

        return uep

    return _with_environment_permissions
```

**Review by khvn26 — APPROVED** [review]
Some typing nitpicks, otherwise LGTM

File: api/tests/unit/features/test_unit_features_views.py (hunk 1 above)
Comment (matthewelwell) [reply api/tests/unit/features/test_unit_features_views.py]: Fixed [here](https://github.com/Flagsmith/flagsmith/pull/2990/commits/64261ae9939e2e37fcd1d0f75273816d5cf76f94)

**Pull request merged**